emfacilities/protocols/protocol_data_counter.py

```diff
@@ -0,0 +1,302 @@
+        inputSet.loadAllProperties()
+        return inputSet
+
+    def _loadOutputSet(self, SetClass, baseName):
+        setFile = self._getPath(baseName)
+
+        if os.path.exists(setFile):
+            outputSet = SetClass(filename=setFile)
+            outputSet.loadAllProperties()
+            outputSet.enableAppend()
+        else:
+            outputSet = SetClass(filename=setFile)
+            outputSet.setStreamState(outputSet.STREAM_OPEN)
+
+        inputs = self.inputImages.get()
+        outputSet.copyInfo(inputs)
+
+        return outputSet
+
+    def _insertNewImageSteps(self, newIds):
+        """ Insert steps to register new images (from streaming)
+        Params:
+            newIds: input images ids to be processed
+        """
+        deps = []
+        stepId = self._insertFunctionStep(self.registerStep, newIds, needsGPU=False,
+                                          prerequisites=[])
+        deps.append(stepId)
+        self.insertedIds.extend(newIds)
+
+        return deps
+
+    def registerStep(self, newIds):
+        self.info('Registering the %d new images' %len(newIds))
+        for imageId in newIds:
+            self.processedIds.append(imageId)
+
+        if not self.isStreamClosed:
+            self.delayRegister()
+            if self.boolTimer.get():
+                self.info('Using timer:')
+                self.timerStep()
+
+    def delayRegister(self):
+        delay = self.delay.get()
+        self.info('Sleeping for delay time: %d' %delay)
+        time.sleep(delay)
+
+    def timerStep(self):
+        endTime = self.lastTimeCheckTimer + timedelta(seconds=self.timeoutSecs)
+        now = datetime.now()
+        remainingTime = (endTime - now).total_seconds()
+
+        if remainingTime <= 0:
+            self.timerOut = True
+            self.info("  timer is consumed terminating protocol.")
+        else:
+            self.info(f"  remaining time: {int(remainingTime)} seconds.")
+            self.timeoutSecs = int(remainingTime)
+
+        # Update the last time check
+        self.lastTimeCheckTimer = now
+
+    # ------------------------- UTILS functions --------------------------------
+    def _getAllDoneIds(self):
+        doneIds = []
+        sizeOutput = 0
+
+        if hasattr(self, OUTPUT):
+            sizeOutput = self.outputSet.getSize()
+            doneIds.extend(list(self.outputSet.getIdSet()))
+
+        return doneIds, sizeOutput
+
+    def getTimeOutInSeconds(self, timeOut):
+        timeOutFormatRegexList = {r'\d+s': 1, r'\d+m': 60, r'\d+h': 3600,
+                                  r'\d+d': 72000}
+        try:
+            return int(timeOut)
+        except Exception:
+            seconds = 0
+        for regex, secondsUnit in timeOutFormatRegexList.items():
+            matchingTimes = re.findall(regex, timeOut)
+            for matchTime in matchingTimes:
+                seconds += int(matchTime[:-1]) * secondsUnit
+
+        return seconds
+
+    def _summary(self):
```

Comment:
Add some details like proportion of your images in the output and a percent of images in the output. Also useful to know that the protocol is running well